watertap/tools/oli_api/credentials.py

```diff
@@ -191,164 +267,138 @@ def set_access_key(self):
-        else:
-            headers = {"authorization": "Bearer " + self.jwt_token}
-        headers.update({"Content-Type": "application/json"})
-        payload = json.dumps({})
-        response = requests.post(self.access_key_url, headers=headers, data=payload)
+        def _set_expiry_timestamp(key_lifetime):
+            """
+            Set expiry date for OLI Cloud access key.
+
+            :param key_lifetime: integer number of days key will be valid (up to 365)
+
+            :return unix_timestamp_ms: unix timestamp (in ms) for when key will expire
+            """
+
+            if key_lifetime < 1 or key_lifetime > 365:
+                key_lifetime = 365
+
+            _logger.debug(f"Maximum key lifetime is 365 days, {key_lifetime} provided.")
+            current_time = datetime.now(timezone.utc)
+            expiry_timestamp = (current_time + timedelta(days=key_lifetime)).timestamp()
+            unix_timestamp_ms = int(expiry_timestamp * 1000)
+            return unix_timestamp_ms
+
+        response = requests.post(
+            self.access_key_url,
+            headers=self.update_headers({"Content-Type": "application/json"}),
+            data=json.dumps({"expiry": _set_expiry_timestamp(key_lifetime)}),
+        )
+
         self.credentials["access_keys"].append(
             json.loads(response.text)["data"]["apiKey"]
         )
-        if not self.test:
-            print(response.text)
+        _logger.info(response.text)
         return response.text
 
     def delete_oliapi_access_key(self, api_key):
         """
         Delete an access key for OLI Cloud.
 
-        :param api_key: The access key to delete.
-        :return string: Response text containing the success message or an error message.
+        :param api_key: The access key to delete
+
+        :return string: Response text containing the success message or an error message
         """
 
-        if self.access_key:
-            headers = {"authorization": "API-KEY " + self.access_key}
-        else:
-            headers = {"authorization": "Bearer " + self.jwt_token}
-        headers.update({"Content-Type": "application/json"})
-        payload = json.dumps({"apiKey": api_key})
-        response = requests.delete(self.access_key_url, headers=headers, data=payload)
-        if not self.test:
-            print(response.text)
+        response = requests.delete(
+            self.access_key_url,
+            headers=self.update_headers({"Content-Type": "application/json"}),
+            data=json.dumps({"apiKey": api_key}),
+        )
+        _logger.info(response.text)
         return response.text
 
-    def login(self):
+    def login(self, refresh=False):
         """
-        Login into user credentials for the OLI Cloud.
+        Log in to OLI Cloud using access key or credentials.
 
-        :return boolean: True on success, False on failure
+        :return status: bool indicating success or failure
```

Comment:
Should "refresh" be in the doc string (even though it is self-explanatory)?

Reply:
@veccp Can you add it? This PR is pretty much ready to go but this is a super minor revision that should be handled in this PR.

Reply:
Yes, I just pushed the change